code/datums/components/directional_attack.dm

```diff
@@ -0,0 +1,155 @@
+		return NONE //We have something in our hand.
+
+/datum/component/directional_attack/proc/carbon_directional_action_checks(mob/living/L)
+	var/mob/living/carbon/attacker = parent
+	. = living_directional_action_checks(L)
+	if(!isnull(.))
+		return
+	if(QDELETED(L))
+		return
+	switch(attacker.a_intent)
+		if(INTENT_HELP, INTENT_GRAB)
+			return NONE
+	
+/datum/component/directional_attack/proc/figure_out_living_target(atom/target)
+	var/clickDir = get_dir(parent, target)
+	var/turf/presumedPos = get_step(parent, clickDir)
+	var/mob/living/L = locate() in presumedPos
+	if(presumedPos == target && L == null)
+		return target
+	return L
+
+/datum/component/directional_attack/proc/living_directional_action(atom/A, params)
+	var/atom/T
+	if(isturf(A))
+		T = A
+	else 
+		return 
+	var/atom/hold = figure_out_living_target(T)
+	if (isnull(hold))
+		return
+	if(isturf(hold))
+		hold = A
+	if(isliving(hold))
+		var/mob/living/L = hold
+		. = living_directional_action_checks(L)
+		if(!isnull(.))
+			return
+	
+	return living_do_directional_action(hold)
+
+/datum/component/directional_attack/proc/human_directional_action(atom/A, params)
+	var/atom/T
+	if(isturf(A))
+		T = A
+	else
+		return
+	var/mob/living/carbon/human/attacker = parent
+	var/atom/hold = figure_out_living_target(T)
+	if (isnull(hold))
+		return
+	if(isturf(hold))
+		hold = A
+	if(isliving(hold))
+		var/mob/living/L = hold
+		. = carbon_directional_action_checks(L)
+		if(!isnull(.))
+			return
+		if(attacker.faction == L.faction)
+			return //FF
+
+	return living_do_directional_action(hold)
+
+/datum/component/directional_attack/proc/xeno_directional_action(atom/A, params)
+	var/atom/T
+	if(isturf(A))
+		T = A
+	else
+		return
+	var/mob/living/carbon/xenomorph/attacker = parent
+	var/atom/hold = figure_out_living_target(T)
+	if (isnull(hold))
+		return	
+	if(isturf(hold))
+		hold = A
+	if(isliving(hold))
+		var/mob/living/L = hold
+		. = carbon_directional_action_checks(L)
+		if(!isnull(.))
+			return
+		if(attacker.issamexenohive(L))
+			return //No more nibbling.
+	
+	return living_do_directional_action(hold)
+
+/datum/component/directional_attack/proc/living_do_directional_action(atom/target)
+	var/mob/living/attacker = parent
+	attacker.UnarmedAttack(target, TRUE)
+	COOLDOWN_START(src, COOLDOWN_DIRECTIONAL_ATTACK, CLICK_CD_MELEE)
+
+//Toggle directional attacks
+/datum/action/directional_attack_toggle
+	name = "Toggle Directional Attacks"
+	var/static/atom/movable/vis_obj/action/bump_attack_active/active_icon = new
+	var/static/atom/movable/vis_obj/action/bump_attack_inactive/inactive_icon = new
+
+
+/datum/action/directional_attack_toggle/New()
+	. = ..()
+	button.overlays.Cut()
+
+/datum/action/directional_attack_toggle/update_button_icon(active)
+	if(isnull(active))
+		return
+	if(active)
+		button.vis_contents -= inactive_icon
+		button.vis_contents += active_icon
+	else
+		button.vis_contents -= active_icon
+		button.vis_contents += inactive_icon
```

Comment:
We should avoid having this as another button.
I'd rather instead this was moved to a verb instead.

Reply:
It is now 1 button for switching between